consensus/state_test.go

```diff
@@ -734,7 +736,7 @@ func TestStateLockPOLSafety1(t *testing.T) {
 	rs := re.(types.EventDataRoundState).RoundState.(*cstypes.RoundState)
 	propBlock := rs.ProposalBlock
 
-	<-voteCh // prevote
+	ensureVote(voteCh, h, r, types.VoteTypePrevote)
```

Comment:
I only did this for this test case because it's the one that broke (awesome btw, tests are working as expected... pubsub or eventbus needs better debug tooling).

ensureVote etc instead of blindly receiving on the channel will help.  For example, it would have helped me tremendously to know that a prevote was expected but a precommit came through.